File: aat-android/src/main/java/ch/bailu/aat/util/ui/ToolTip.java
package ch.bailu.aat.util.ui;

import android.graphics.Color;
import android.os.Build;
import android.view.View;
import android.widget.TextView;

import ch.bailu.aat_lib.map.MapColor;

public class ToolTip {
    public static void set(View view, Integer resID) {
        if (Build.VERSION.SDK_INT >= 26) {
            view.setTooltipText(new ToolTipRes(view.getContext(), resID).getToolTip());
        } else {
            new ToolTipLongClick(view, new ToolTipRes(view.getContext(), resID));
        }
    }


    public static void set(View view, String toolTip) {
        if (Build.VERSION.SDK_INT >= 26) {
            view.setTooltipText(toolTip);
        } else {
            new ToolTipLongClick(view, new ToolTipString(toolTip));
        }

    }

    public static void themeify(TextView v) {
        v.setBackgroundColor(MapColor.LIGHT);
        v.setTextColor(Color.BLACK);
        AppTheme.padding(v,10);

    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/util/ui/tooltip/ToolTip.kt
package ch.bailu.aat.util.ui.tooltip

import android.graphics.Color
import android.os.Build
import android.view.View
import android.widget.TextView
import ch.bailu.aat.util.ui.theme.AppTheme
import ch.bailu.aat_lib.map.MapColor

object ToolTip {

    @JvmStatic
    fun set(view: View, resID: Int) {
        if (Build.VERSION.SDK_INT >= 26) {
            view.tooltipText = ToolTipRes(view.context, resID).toolTip
        } else {
            ToolTipLongClick(view, ToolTipRes(view.context, resID))
        }
    }

    @JvmStatic
    fun set(view: View, toolTip: String) {
        if (Build.VERSION.SDK_INT >= 26) {
            view.tooltipText = toolTip
        } else {
            ToolTipLongClick(view, ToolTipString(toolTip))
        }
    }

    @JvmStatic
    fun themeify(v: TextView) {
        v.setBackgroundColor(MapColor.LIGHT)
        v.setTextColor(Color.BLACK)
        AppTheme.padding(v, 10)
    }
}
